architecture/2025-02-user-defined-resource-type-feature-spec.md

```diff
@@ -0,0 +1,1212 @@
+     * Update the local bicepconfig.json with the file location
+   * – or –
+     * Create the Bicep extension on the workstation and store in on a shared drive 
+     * Update the local bicepconfig.json
+3. Repeat this process for each new or modified resource type
+
+**To Be User Experience**
+
+When the platform engineer creates or modifies a resource type via `rad resource-type create` Radius will:
+
+1. Compile a Bicep extension (this is performed first to catch compilation errors early)
+2. Create or update the resource type in Radius
+3. Store the Bicep extension within the Radius control plane which has the appropriate enhancements to support Bicep extensions
+
+When the resource type is deleted, the associated Bicep extension is deleted as well.
+
+When the developer executes rad CLI commands Radius will:
+
+* `rad workspace create` – Creating a new workspace will create a bicepconfig.json in the working directory which has a reference to each Radius-hosted Bicep extension (if bicepconfig.json exists in the working directory, update the file to ensure it has each of the Radius-hosted Bicep extensions)
+* `rad workspace update` – This is a new command which will refresh the local configuration with the latest Bicep extensions
+* `rad workspace switch`, `rad run`, or `rad deploy` – If the bicepconfig.json file is out of date relative to the Radius-hosted version, the command will fail and instruct the user to run ``rad workspace update`.
+
+## Scenario 2 – Authoring recipes 
+
+### **User Story 5 – Composite resources** 
+
+As a platform engineer, I need to define a resource type which is composed of many resources. Some of the embedded resources will be core Radius resources but others may be other resource types I have created.
+
+**Summary**
+
+The goal of user-defined resource types is to enable users and the broader Radius community to build a library of resource types which model cloud-native application components rather than cloud infrastructure resources. Some organizations will choose to model their resources with granular resource types such as containers and databases, while others will only offer abstracted resource types such as a microservice in their resource catalog. These abstract resource types will be composed of many resources just an an application is composed of many resources. 
+
+A microservice, for example, could be composed of an application container, an in-memory cache, a reverse proxy, and a container security sidecar. In order to model the microservice resource type, the platform engineer needs the ability to specify a schema for the resource type, as well as the resources that compose the microservice. What makes this user story unique compared to those previously discussed is that the resources composing this microservice are other Radius resource types—not only infrastructure. 
+
+**`webservice-resource-type.bicep`**:
+
+`````yaml
```

Comment:
is this illustrating that composite types can be defined using YAML? would we recommend users against this and point them to defining composite types via Bicep/Terraform Recipes?